lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -844,15 +815,112 @@ private void indexObjectWrapperFn(final List<Object> input)
         
         private List<ObjectData> getObjectsByIds(final Set<GUID> guids)
                 throws RetriableIndexingException {
-            kbasesearchengine.search.PostProcessing pp = 
-                    new kbasesearchengine.search.PostProcessing();
-            pp.objectData = false;
-            pp.objectKeys = true;
-            try {
-                return indexingStorage.getObjectsByIds(guids, pp);
-            } catch (IOException e) {
-                throw new RetriableIndexingException(ErrorType.OTHER, e.getMessage(), e);
+
+            final List<ObjectData> data = new ArrayList<>();
+
+            // get object from data source
+            for (GUID guid: guids) {
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
+                    tempFile = File.createTempFile("ws_srv_response_", ".json");
+                } catch (IOException ex) {
+                    throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage(), ex);
+                }
+
+                try {
+                    final EventHandler handler = getEventHandler(guid);
+                    final Iterator<ResolvedReference> refs = handler.resolveReferences(null, guids).iterator();
+                    while (refs.hasNext()) {
+                        final ResolvedReference ref = refs.next();
+
+                        // make a copy to avoid mutating the caller's path
+                        final LinkedList<GUID> newRefPath = new LinkedList<>();
+                        newRefPath.add(guid);
+                        final SourceData obj = handler.load(newRefPath, tempFile.toPath());
+
+                        final List<ObjectTypeParsingRules> filteredParsingRules = getFilteredSortedParsingRules(ref);
+
+                        for (final ObjectTypeParsingRules rule : filteredParsingRules) {
```

Comment:
I have fixed this. But it is now possible that an object will not get returned for a guid if the parsing rules cannot be found. I don't like the silent failure but it seems to be the underlying theme of the code for this case. I tried throwing an indexing exception but that exception has a protected constructor in a different package. It doesn't make sense to me. I should be able to generate an indexing exception from the indexer worker.

Reply:
I have fixed this by throwing an exception. But it broke your guidNotFound test since the new exception gets thrown before the guidNotFound exception. I modified that test some to accommodate for this change.